domain/src/main/java/org/oppia/android/domain/snackbar/SnackbarController.kt

```diff
@@ -0,0 +1,58 @@
+package org.oppia.android.domain.snackbar
+
+import androidx.annotation.StringRes
+import org.oppia.android.util.data.AsyncDataSubscriptionManager
+import org.oppia.android.util.data.DataProvider
+import org.oppia.android.util.data.DataProviders
+import java.util.LinkedList
+import java.util.Queue
+import javax.inject.Inject
+import javax.inject.Singleton
+
+private const val SNACKBAR_REQUEST_PROVIDER_ID = "snackbar_request_provider_id"
+
+@Singleton
+class SnackbarController @Inject constructor(
+  private val dataProviders: DataProviders,
+  private val asyncDataSubscriptionManager: AsyncDataSubscriptionManager,
+) {
+
+  private val _snackbarRequestQueue: Queue<SnackbarRequest.ShowSnackbar> = LinkedList()
+
+  val snackbarRequestQueue: Queue<SnackbarRequest.ShowSnackbar>
+    get() = _snackbarRequestQueue
+
+  fun getCurrentSnackbar(): DataProvider<SnackbarRequest> {
```

Comment:
Please add KDocs and a test for this method.

Reply:
okay